### ISSUE TITLE
searchAuthorities OC Alfresco Implementation does not honor subgroups

### ISSUE THREAD

**Issue by dgrumieaux** [tsgrp/HPI#1102]
Given a user Bob is a member of subgroup B, and given subgroup B is a member of group A:

I would expect searchAuthorities, when asking it to return members of group A, to return Bob. The Alfresco implementation does not do this today.

This impacts the Wizard delegate and reassign actions.

**Comment by gsteimer** [comment]
@dgrumieaux @nwoodbine - Fwiw, it looks like the code is _trying_ to get users in subgroups:

**AlfrescoEmbUserImpl (690)**
```
Set<String> containedAuths = serviceRegistry.getAuthorityService().getContainedAuthorities(type, criteria.getChildOfAuthority(), false);
```

That last parameter is `immediate`.  I would think setting it to false would get users at any depth.  Javadocs from Alfresco source on that param:

```
     * @param immediate -
     *            if true, limit the depth to just immediate child, if false
     *            find authorities at any depth
```

**Comment by gsteimer** [comment]
@dgrumieaux @nwoodbine - Ok, I'm adding a new `usersInGroup` picklist to AlfrescoEmbPicklistImpl, and I tried the exact code above.  Here's what I'm seeing:

```
Set<String> containedAuths = serviceRegistry.getAuthorityService().getContainedAuthorities(AuthorityType.USER, queryParams[1], false);
```
Alfresco Groups:
![image](https://cloud.githubusercontent.com/assets/1377228/15576711/c060bd20-231e-11e6-9e94-97f905c97709.png)

HPI Picklist in the admin:
![image](https://cloud.githubusercontent.com/assets/1377228/15576721/d248d7c0-231e-11e6-9a6e-e8c16a074dd1.png)

Long story short - it looks like the alfresco code _is_ working properly.  Based on discussions with @dgrumieaux, we're thinking the code isn't running the way we think it is...

**Comment by dgrumieaux** [comment]
I think there's a query being built up that's utilizing the PARENT
qualifier, whereas our code should use the snippet that @gsteimer's post
uses...

**Comment by gsteimer** [comment]
Fixed in OC Rev 15856
